.github/workflows/scan-rock.yaml

```diff
@@ -36,25 +34,25 @@ jobs:
           remove-android: 'true'
           remove-codeql: 'true'
 
-      - name: Check out code
-        uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.source-branch }}
 
-      - name: Build ROCK
-        uses: canonical/craft-actions/rockcraft-pack@main
-        id: rockcraft
+      - uses: actions/download-artifact@v3
         with:
-          path: ./${{ inputs.rock-dir }}
+          name: ${{ inputs.rock-dir }}
+          path: ${{ inputs.rock-dir }}/
 
-      - name: Copy ROCK to Docker
+      - name: Export ROCK to Docker
         id: rock_in_docker
         run: |
           NAME=$(yq eval ".name" rockcraft.yaml)
           VERSION=$(yq eval ".version" rockcraft.yaml)
-          ARCH=$(yq eval ".platforms | keys" rockcraft.yaml | awk -F ' ' '{print $2}')
-          ROCK="${NAME}_${VERSION}_${ARCH}"
-          sudo skopeo --insecure-policy copy oci-archive:$ROCK.rock docker-daemon:$ROCK:$VERSION
-          echo "Copied $ROCK:$VERSION to Docker"
-          echo "image=$ROCK:$VERSION" >> "$GITHUB_OUTPUT"
+          DOCKER_IMAGE=$NAME:$VERSION
+          ROCK=$(echo ${{ inputs.path-to-rock-file }} | sed 's:.*/::')
+          sudo skopeo --insecure-policy copy oci-archive:$ROCK docker-daemon:$DOCKER_IMAGE
+          echo "image=$DOCKER_IMAGE" >> "$GITHUB_OUTPUT"
```

Comment:
I think we need `inputs.rock-dir` to call download-artifact since that's how we 've named the artifact and we need to name it like this instead of just `rock`, since we can have many `.rock` artifacts in one workflow if many rocks are modified.

Reply:
Agreed, removed all inputs and added only `rock-filename` which is also the name of the artifact uploaded by `build` job.